Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -136,17 +145,29 @@ namespace prime {
     handle_beam_visor_switch(prime_one_beams, prime_one_visors);
 
     // Allows freelook in grapple, otherwise we are orbiting (locked on) to something
-    if (read32(mp1_static.orbit_state_address) != ORBIT_STATE_GRAPPLE &&
-      read8(mp1_static.lockon_address)) {
+    bool locked = (read32(mp1_static.orbit_state_address) != ORBIT_STATE_GRAPPLE &&
+                   read8(mp1_static.lockon_address));
+    
+    if (locked) {
       write32(0, mp1_static.yaw_vel_address);
-      return;
+      // Calculate the pitch based on the XF matrix of player so that we always
+      // write out the pitch even while locked onto the target, this prevents
+      // the view from snapping back after releasing a lock-on
+      calculate_pitch_locked();
+    } else {
+      calculate_pitch_delta();
     }
 
     // I write to two locations here to control the pitch, the rate of turning
     // has been unlocked already
-    calculate_pitch_delta();
     writef32(pitch, mp1_static.pitch_address);
     writef32(pitch, mp1_static.pitch_goal_address);
+    if (locked) {
```

Comment:
Assuming pitch will be identical to what the game has currently stored, I would move the pitch writes to the `else` block since that is the only case it should be written. Ideally you should return before writing pitch when locked if it is going to be writing the exact same value anyway.

Reply:
Unfortunately if you only write the pitch while not locked then you get this 1-frame snap where your view snaps back to the previous location.

This happens because the frame where you release lock is processed and rendered before this mod is run, which means by the time we are able to start writing the pitch again the game has already rendered a frame without locking and in turn causes a 1-frame stutter-like snap immediately after locking.

One solution would be to have a better hookpoint to run this mod, one where the input has been processed but the frame hasn't been rendered yet.

Short of a better hook the only way to handle the 1-frame snap is to just write out the pitch throughout the entirety of being locked on, even if it's not actually visible while locked on. That way the very frame the lock is released it will render the correct viewangle.